android-components/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/OnDiskSitePermissionsStorage.kt

```diff
@@ -74,12 +82,17 @@ class OnDiskSitePermissionsStorage(
     /**
      * Finds all SitePermissions that match the [origin].
      * @param origin the site to be used as filter in the search.
+     * @param private indicates if the [SitePermissions] belongs to a private session.
      */
-    override suspend fun findSitePermissionsBy(origin: String, includeTemporary: Boolean): SitePermissions? {
+    override suspend fun findSitePermissionsBy(
+        origin: String,
+        includeTemporary: Boolean,
+        private: Boolean,
+    ): SitePermissions? {
         return database
             .sitePermissionsDao()
             .getSitePermissionsBy(origin)
-            ?.toSitePermission()
+            ?.toSitePermission()?.resetIfPrivate(private)
```

Comment:
If it's a private session, why not just early return instead of hitting the db and then returning a newly created `SitePermissions` object?

Reply:
Good catch!